adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -242,6 +255,54 @@
             return adapter.login(options);
         }
 
+        function generateRandomData(len) {
+            // use web crypto APIs if possible
+            var array = null;
+            var crypto = window.crypto || window.msCrypto;
+            if (crypto && crypto.getRandomValues && window.Uint8Array) {
+                array = new Uint8Array(len);
+                crypto.getRandomValues(array);
+                return array;
+            }
+
+            // fallback to Math random
```

Comment:
@thomasdarimont @mposolda `Math.random` is not considered cryptographically secure and I don't think is a good idea use it as fallbac. If Webcrypto is not supported in the browser, I believe would be better to provide a warning message like "Not supported" or something similar.

Reply:
Yes, `Math.random` does not provide secure randomness, but isn't a little less random 96 char long `codeVerifier` string better than no `codeVerifier` at all? 
I see your point of just providing secure options, but a `console.warn(..)` message about not strong enough crypto might not help the user very much, because they usually don't look in the developer console :)

Further more this would only happen for old browsers or Opera Mini (see. https://caniuse.com/#search=getRandomValues). 

Btw. the length of 96 chars was also just a guess along the recommended size of > 43 and < 128 (see: https://tools.ietf.org/html/rfc7636).

Reply:
IMO it will be good to use the most secured function available in current browser and fallback to the less secure alternative if the more secure is not available.

The `Math.random` is already used for createUUID BTV. I don't think we need to skip functionality if more secure alternative is not available, I don't see the fact that `Math.random` is not 100% cryptographically secure as a major potential for attack vectors.

Will be good to have unified "random" function for all the functionality in the JS library, which don't need whole alphabet as an argument.